core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -0,0 +1,224 @@
+            PersonIdent committer = new PersonIdent(defaultCommitter, Date.from(plugin.getReleaseTimestamp().minusDays(1).toInstant()));
+
+            git.add().addFilepattern("pom.xml").call();
+            git.commit().setMessage("Imports pom.xml file").setSign(false).setCommitter(committer).call();
+
+            final HasUnreleasedProductionChangesProbe probe = getSpy();
+            final ProbeResult result = probe.apply(plugin, ctx);
+
+            assertThat(probe.apply(plugin, ctx))
+                .usingRecursiveComparison()
+                .comparingOnlyFields("id", "status")
+                .isEqualTo(result.success(HasUnreleasedProductionChangesProbe.KEY, ""));
+        }
+    }
+
+    @Test
+    void commitOnReadmeFileAfterReleaseDateShouldReturnSuccess() throws IOException, GitAPIException {
+        final Path repository = Files.createTempDirectory("test-foo-bar");
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+
+        ZonedDateTime releaseTimestamp = ZonedDateTime.now();
+        when(plugin.getReleaseTimestamp()).thenReturn(releaseTimestamp);
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, "")
+        ));
+        when(ctx.getScmRepository()).thenReturn(repository);
+
+        try (Git git = Git.init().setDirectory(repository.toFile()).call()) {
+
+            final Path pom = Files.createFile(repository.resolve("README.md"));
+
+            // creating commit
+
+            PersonIdent defaultCommitter = new PersonIdent(git.getRepository());
+            PersonIdent committer = new PersonIdent(defaultCommitter, Date.from(plugin.getReleaseTimestamp().plusDays(1).toInstant()));
+
+            git.add().addFilepattern("README.md").call();
+            git.commit().setMessage("Updated README.md file").setSign(false).setCommitter(committer).call();
+
+            final HasUnreleasedProductionChangesProbe probe = getSpy();
+            final ProbeResult result = probe.apply(plugin, ctx);
+
+            assertThat(probe.apply(plugin, ctx))
+                .usingRecursiveComparison()
+                .comparingOnlyFields("id", "status")
+                .isEqualTo(result.success(HasUnreleasedProductionChangesProbe.KEY, ""));
+
+
+        }
+    }
+
+    @Test
+    void checkThatCommitOnSrcPathBeforeReleaseDateReturnsSuccess() throws IOException, GitAPIException {
+        final Path repository = Files.createTempDirectory("test-foo-bar");
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+
+        ZonedDateTime releaseTimestamp = ZonedDateTime.now();
+        when(plugin.getReleaseTimestamp()).thenReturn(releaseTimestamp);
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, "")
+        ));
```

Comment:
```suggestion
        when(plugin.getDetails()).thenReturn(Map.of(
            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
        ));
```

Reply:
Currently the test shows an error:
```
java.lang.AssertionError: 
Expecting actual:
  ProbeResult[id=unreleased-production-changes, message=unreleased-production-changes does not meet the criteria to be executed on null, status=ERROR, timestamp=2023-05-18T11:26:33.365796514+05:30[Asia/Kolkata]]
to be equal to:
  ProbeResult[id=unreleased-production-changes, message=, status=SUCCESS, timestamp=2023-05-18T11:26:33.639163896+05:30[Asia/Kolkata]]
when recursively comparing field by field, but found the following difference:

field/property 'status' differ:
- actual value  : ERROR
- expected value: SUCCESS
```

and the reason we're seeing this error is explained here: https://github.com/jenkins-infra/plugin-health-scoring/pull/298#discussion_r1187688931